cuda_core/tests/test_module.py

```diff
@@ -245,3 +246,14 @@ def test_num_args_error_handling(deinit_all_contexts_function, cuda12_prerequisi
     with pytest.raises(CUDAError):
         # assignment resolves linter error "B018: useless expression"
         _ = krn.num_arguments
+
+
+def test_module_serialization_roundtrip(get_saxpy_kernel):
+    _, objcode = get_saxpy_kernel
+    result = pickle.loads(pickle.dumps(objcode))  # nosec B403, B301
+
+    assert isinstance(result, ObjectCode)
+    assert result._handle is None
```

Comment:
I'd lean towards removing that from here, otherwise this unit test mixes exercising the serialization roundtrip and unrelated implementation details (in this case related to the lazy loading feature).